compiler/src/iree/compiler/Codegen/Common/GPU/GPUMaterializeEncoding.cpp

```diff
@@ -175,15 +228,57 @@ struct GPUSetEncodingOpLoweringConversion
       assert(packedShape == targetShape);
     }
 
-    // TODO(lialan): create expand_shape. Take LHS as an example:
-    // 16x4xf32 -> 16x1x4x1. Because the vector size used in the intrinsic is
-    // 1x1.
-    // For C-Matrix (i.e., ACC), it is 16x16xf32 -> 4x4x16x1xf32. Because the
-    // vector size is 4x1.
+    // Check that the dimensions of the matrix can be divided by the tile shape,
+    // if not then bail out.
+    auto sourceShape = packOp->getResult().getType().getShape();
+    assert(sourceShape.size() == 4);
+    // inner most two dimensions must be divisible.
+    if (sourceShape[2] % innerTiles[0] != 0 ||
+        sourceShape[3] % innerTiles[1] != 0) {
+      return failure();
+    }
```

Comment:
The check is not needed. The inner shape of the pack op must be innerTiles by definition.

Reply:
removed